lib/kvdb/kvdb_meta.c

```diff
@@ -578,6 +578,16 @@ kvdb_meta_deserialize(struct kvdb_meta *const meta, const char *const kvdb_home)
         err = merr(EPROTO);
         goto out;
     }
+    if (version_val < 0) {
+        log_err("'version' key in %s/kvdb.meta is negative", kvdb_home);
+        err = merr(EPROTO);
+        goto out;
+    }
+    if (version_val > UINT_MAX) {
+        log_err("'version' key in %s/kvdb.meta is negative", kvdb_home);
+        err = merr(EPROTO);
+        goto out;
+    }
```

Comment:
Seems really really weird that we're using a double for this...

Reply:
Second error message says "is negative", but it's actually positive and too large.  You could use one message for all cases:  "Invalid version: must be integer value between 0 and UINT_MAX".

Reply:
cJSON returns a double. I have no control